pkg/gateway/operations/putobject.go

```diff
@@ -325,3 +338,20 @@ func handlePut(w http.ResponseWriter, req *http.Request, o *PathOperation) {
 	o.SetHeader(w, "ETag", httputil.ETag(blob.Checksum))
 	w.WriteHeader(http.StatusOK)
 }
+
+func (o *PathOperation) checkIfAbsent(req *http.Request) (bool, error) {
```

Comment:
I find this function confusing because: 
It does 3 different things: 
- It checks and validates the the header value  
- Get's the value from catalog. 
- Returns `allowOverride` indicator for the real code somewhere else that checks if absent. 

**Now that is confusing because it's a "lie" - it only's only partially checking ifAbsent as optimization, since the real check performed later and also does input validation.** 

I would prefer something much more explicit like a function that extracts the header and validates it, then inline check if object exist: 

```go
// checkIfAbsent sets allowOverwrite and validates the header value if set 
allowOverwrite, err := o.checkIfAbsent(req)
if err != nil { 
    // ...
}
if !allowOverwrite { 
    // first check if object exist as optimization to save resources 
    _, err := o.Catalog.GetEntry(req.Context(), o.Repository.Name, o.Reference, o.Path, catalog.GetEntryParams{}) 
    // hadle if err != nil ...
}
```